app/scripts/metamask-controller.js

```diff
@@ -2043,7 +2043,11 @@ export default class MetamaskController extends EventEmitter {
    * @param {boolean} active - True if price data should be getting fetched.
    */
   set isClientOpenAndUnlocked (active) {
-    this.tokenRatesController.isActive = active
```

Comment:
This setter is called pretty often - multiple times when opening the popup, for example. We don't want to trigger new exchange rate updates multiple times during each UI initialization.

I'd suggest starting and stopping this in the `controllerConnectionChanged` handler instead. We use that for a couple of other controllers, where we want them "active" when _some_ UI is open, but stopped if everything is closed. This seems like what we want here too.

Reply:
thanks that makes sense. This resulted in removing some no longer necessary listener/methods when you do your re-review can you confirm that this is safe?

Reply:
Sure! Yeah I noticed looking into this that there would now be a lot of things no longer needed.